src/main/java/io/appium/java_client/remote/AppiumCommandExecutor.java

```diff
@@ -38,67 +39,132 @@
 import org.openqa.selenium.remote.Response;
 import org.openqa.selenium.remote.ResponseCodec;
 import org.openqa.selenium.remote.codec.w3c.W3CHttpCommandCodec;
-import org.openqa.selenium.remote.http.ClientConfig;
 import org.openqa.selenium.remote.http.HttpClient;
 import org.openqa.selenium.remote.http.HttpRequest;
 import org.openqa.selenium.remote.http.HttpResponse;
 import org.openqa.selenium.remote.service.DriverService;
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import java.io.IOException;
 import java.lang.reflect.Field;
 import java.net.ConnectException;
+import java.net.MalformedURLException;
 import java.net.URL;
-import java.time.Duration;
 import java.util.Map;
 import java.util.Optional;
 import java.util.UUID;
 
 public class AppiumCommandExecutor extends HttpCommandExecutor {
     // https://github.com/appium/appium-base-driver/pull/400
     private static final String IDEMPOTENCY_KEY_HEADER = "X-Idempotency-Key";
-    private static final Duration DEFAULT_READ_TIMEOUT = Duration.ofMinutes(10);
 
     private final Optional<DriverService> serviceOptional;
 
-    private AppiumCommandExecutor(Map<String, CommandInfo> additionalCommands, DriverService service,
-                                  URL addressOfRemoteServer,
-                                  HttpClient.Factory httpClientFactory,
-                                  ClientConfig clientConfig) {
+    private final HttpClient.Factory httpClientFactory;
+
+    private final AppiumClientConfig appiumClientConfig;
+
+    private static class DirectConnect {
```

Comment:
I would say it is ok to have this class public. Also, you could use [lombok](https://projectlombok.org/) for this purpose, which greatly reduces the amount of bilerplate code for POJOs

Reply:
also having it public simplifies unit testing

Reply:
Moved to a new public class file. I'll try the lambok later as another pr after the UA change (So far, I've left a TODO comment)